app/routes/moon_routes.py

```diff
@@ -0,0 +1,69 @@
+from app import db
+from app.models.moon import Moon
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, abort, make_response, request
+from .validate_routes import validate_model, validate_moon_user_input, validate_planet_user_input
+
+moons_bp = Blueprint("moons_bp", __name__, url_prefix = "/moons")
+
+# Get all moons info 
+# /moons
+# Return JSON list
+@moons_bp.route("", methods = ["GET"])
+def get_all_moons_query():
+    moon_query = Moon.query
+    # Filtering by moon name (return all records which name contains planet_name_query)
+    moon_name_query = request.args.get("name")
+    
+    if moon_name_query:
+        moon_query = moon_query.filter(Moon.name.ilike(f"%{moon_name_query}%"))
+    
+    # Sorting by moon name
+    sort_by_name_query = request.args.get("sort_by_name")
+    
+    if sort_by_name_query == "desc":
+        moon_query = moon_query.order_by(Moon.name.desc()).all()
+    elif sort_by_name_query == "asc":
+        moon_query = moon_query.order_by(Moon.name).all() 
+        
+    # Sorting by moon size
+    moon_sort_size_query = request.args.get("sort_by_size")
+    
+    if moon_sort_size_query == "desc":
+        moon_query = moon_query.order_by(Moon.size.desc()).all()
+    elif sort_by_name_query == "asc":
+        moon_query = moon_query.order_by(Moon.size).all()
+
+    # Build response
+    moon_response = []
+    for moon in moon_query:
+        moon_response.append(moon.to_dict())
```

Comment:
moon_response.append() = [moon.to_dict() for moon in self.moons]. Is it right?

Reply:
moon_response+= [moon.to_dict() for moon in moon_query]

Reply:
Nice, you're about there! We would use `=` since the comprehension creates the whole list we want, we don't need to append to, or extend a list that already exists.